sdk/core/core/CMakeLists.txt

```diff
@@ -4,8 +4,41 @@
 cmake_minimum_required (VERSION 3.10)
 set(TARGET_NAME "az_core")
 
-project (${TARGET_NAME} LANGUAGES C)
+# Read in az_version.h
+#   The build will fail with an error if the file is not found or improperly formatted
+set(AZ_SDK_VERSION_H_FILE "${CMAKE_CURRENT_SOURCE_DIR}/inc/az_version.h")
 
+if(EXISTS ${AZ_SDK_VERSION_H_FILE})
+    file (STRINGS ${AZ_SDK_VERSION_H_FILE} VERSION_H_CONTENT)
+    message(STATUS "Read version from ${AZ_SDK_VERSION_H_FILE}")
+
+    #Find and read the AZ_SDK_VERSION_STRING define
+    string(REGEX MATCHALL "#define AZ_SDK_VERSION_STRING \"([0-9]+)\.([0-9]+)\.([0.9]+)([^\"]*)\"" VERSION_PARTS ${VERSION_H_CONTENT})
```

Comment:
Given that the optional `CMAKE_MATCH_4` expects -{some-non-quote-containing-string}, we should make sure the regex only allows that.

That means, that if any character exists after 0.0.0, it better be a hyphen followed by 1 or more non-quote characters. Right now, `0.0.0preview1` would work and `PRERELEASE_SUBSTRING` would be set to `review1`, which isn't what we want.

Reply:
Added stronger checks to the prerelease portion